polaris/model_step.py

```diff
@@ -70,14 +70,18 @@ def __init__(self, task, name, subdir=None, ntasks=None,
 
         Parameters
         ----------
-        task : polaris.Task
-            The task this step belongs to
+        component : polaris.Component
+            the component that the step belongs to
 
         name : str
             The name of the step
 
         subdir : str, optional
-            the subdirectory for the step.  The default is ``name``
+            the subdirectory for the step.  If neither this nor ``indir``
+             are provided, the directory is the ``name``
+
+        indir : str, optional
+            the directory the step is in, to which ``name`` will be appended
```

Comment:
The `subdir` is the directory under the components work directory where the step exists.  The `indir` is one level up from that, with the `name` of the step appended to it to get the `subdir`.  Before shared steps, the default behavior was to take the task's subdirectory and append the name of the step to get the default `subdir` for the step.  The easiest way to efficiently recreate this default behavior without steps having ot know about tasks is to have `indir` be the task's subdirectory.  But there's no reason that it has to be, it could be any convenient directory that is one up from the step's subdirectory, as long as it makes sense for the step's `name` to be appended to it to get the `subdir` for the step.  I don't think there are currently any examples where `indir` is anything other than the parent task's `subdir`.